### ISSUE TITLE
Variable name conflict when two modules declare exports with the same name

### ISSUE THREAD

**Issue by stevenfan** [rollup/rollup#1384]
Repro:
https://rollupjs.org/repl?version=0.41.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyME1vZHVsZUElMjBmcm9tJTIwJTVDJTIyLiUyRm1vZHVsZV9hLmpzJTVDJTIyJTNCJTVDbiU1Q25mdW5jdGlvbiUyMGYoKSUyMCU3QiU1Q24lMjAlMjB2YXIlMjBmb28lMjAlM0QlMjBNb2R1bGVBLmZvbygpJTNCJTVDbiUyMCUyMHJldHVybiUyMGZvbyUzQiU1Q24lN0QlNUNuY29uc29sZS5sb2coZigpKSUzQiUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtb2R1bGVfYS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAqJTIwYXMlMjBNb2R1bGVCJTIwZnJvbSUyMCU1QyUyMi4lMkZtb2R1bGVfYiU1QyUyMiUzQiU1Q24lNUNuZXhwb3J0JTIwZnVuY3Rpb24lMjBmb28oKSUyMCU3QiU1Q24lMjAlMjByZXR1cm4lMjAlNUMlMjJhJTVDJTIyJTNCJTVDbiU3RCUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtb2R1bGVfYi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBmdW5jdGlvbiUyMGZvbygpJTIwJTdCJTVDbiUyMCUyMHJldHVybiUyMCU1QyUyMmIlNUMlMjIlM0IlNUNuJTdEJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJtb2R1bGVJZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

This can result in a line like `var foo = Module.foo();` being rewritten to `var bar = bar();`, which is always wrong.

**Comment by stevenfan** [comment]
It seems like the problem is that names imported into a module don't correctly consider changes (e.g. during a previous deshadow pass of another module)

```diff
index f0f5f7a..9a1f9ee 100644
--- a/src/ast/scopes/ModuleScope.js
+++ b/src/ast/scopes/ModuleScope.js
@@ -21,7 +21,7 @@ export default class ModuleScope extends Scope {
                        if ( specifier.module.isExternal ) return;

                        specifier.module.getExports().forEach( name => {
-                               names.add( name );
+                               names.add( specifier.module.traceExport(name).name );
                        });

                        if ( specifier.name !== '*' ) {
```

**Comment by Rich-Harris** [comment]
Thanks, this is fixed in 0.42
